server/src/main/java/org/opensearch/index/shard/IndexShard.java

```diff
@@ -503,6 +509,33 @@ public boolean shouldSeedRemoteStore() {
         return shardMigrationState == REMOTE_MIGRATING_UNSEEDED;
     }
 
+    /**
+     * To be delegated to {@link ReplicationTracker} so that relevant remote store based
+     * operations can be ignored during engine migration
+     * <p>
+     * Has explicit null checks to ensure that the {@link ReplicationTracker#invariant()}
+     * checks does not fail during a cluster manager state update when the latest replication group
+     * calculation is not yet done and the cached replication group details are available
+     */
+    public Function<String, Boolean> isShardOnRemoteEnabledNode = (shardId) -> {
+        DiscoveryNode discoveryNode = this.discoveryNodes.get(shardId);
```

Comment:
Sorry how is discoveryNodes getting the shard information?

Reply:
From `ReplicationTracker`. We feed the current assigned node ID for each ShardRouting in the replication tracker to this method.

Reply:
But how do we disambiguate between primary or any replica copy for the shard id?

Reply:
This is only used in ReplicationTracker to check if a checkpoint against an allocationId needs to be replicated to the replicas. We don't need to distinguish between a primary/replica copy here. Depending on whether the checkpoint is replicated, PRRLs and other actions which needs to be tracked for docrep based shard copies are decided.